internal/gitea/repo.go

```diff
@@ -0,0 +1,190 @@
+
+	releases, _, err := r.client.ListReleases(r.org, r.Details.Name, opts)
+	if err != nil {
+		return errors.New("error listing releases for repo")
+	}
+
+	for _, rel := range releases {
+		r.Details.Releases = append(r.Details.Releases, &repos.Release{
+			ID:         rel.ID,
+			Version:    rel.TagName,
+			Timestamp:  rel.PublishedAt.Unix(),
+			Title:      rel.Title,
+			Body:       renderReleaseBody(rel.Note, r),
+			URL:        rel.URL,
+			CompareURL: fmt.Sprintf("%s/compare/%s...%s", r.Details.URL, rel.TagName, r.defaultBranch),
+		})
+	}
+
+	return nil
+}
+
+// processCommitsSinceRelease calculates the number of commits that have occurred on the default
+// branch of the repository since the last release, and populates the information in r.Details.
+func (r *Repository) processCommitsSinceRelease() error {
+	// This does not currently handle monorepos - see note about releases in
+	// `processReleases`. If there are releases in Gitea with a monorepo, then
+	// this function maybe needs to restrict the commits to ones where the tree
+	// overlaps - but maybe we actually want to know how many commits overall,
+	// because we want to count things like common code being adjusted.
+
+	// Add the commit delta between last release and default branch.
+	latestRelease := r.Details.Releases[len(r.Details.Releases)-1]
+	opts := gitea.ListCommitOptions{ListOptions: gitea.ListOptions{PageSize: giteaReleasesPerRepo}, SHA: latestRelease.Version, Path: ""}
+	commits, _, err := r.client.ListRepoCommits(r.org, r.Details.Name, opts)
+	if err != nil {
+		return errors.New("error getting commits since release")
+	}
+
+	r.Details.NewCommits = len(commits)
+
+	return nil
+}
+
+// processCommits fetches the latest 3 commits to a repository and populates them into the repo
+// struct in the case that there are no releases identified.
+func (r *Repository) processCommits() error {
+	// This does not currently handle monorepos. It's not clear which commits
+	// should be counted in this case - only ones that have a tree that overlaps
+	// with the subfolder? All commits, as now? If there's common code, then
+	// all commits is probably truest to the single-repo meaning, but if there
+	// are commits that are only in a separate charm, then those probably should
+	// not be included.
+
+	opts := gitea.ListCommitOptions{ListOptions: gitea.ListOptions{PageSize: giteaReleasesPerRepo}, SHA: r.defaultBranch, Path: ""}
+
+	// If there are no releases, get the latest commit instead.
+	commits, _, err := r.client.ListRepoCommits(r.org, r.Details.Name, opts)
+	if err != nil {
+		return errors.New("error listing commits for repository")
+	}
+
+	// Iterate over the commits and append them to r.Details.Commits
+	for _, commit := range commits {
+		// Some commits don't have an author.
+		var name string
+		if commit.Author == nil {
+			name = ""
+		} else {
+			name = commit.Author.FullName
+		}
```

Comment:
Empty string is the zero value for strings (so string variables are initialized to `""`), and I would simplify slightly with `:=`:

```go
name := ""
if commit.Author != nil {
    name = commit.Author.FullName
}
```